test/functional/save_pages_controller_test.rb

```diff
@@ -0,0 +1,108 @@
+require "test_helper"
+require "gds_api/test_helpers/account_api"
+
+class SavePagesControllerTest < ActionController::TestCase
+  include GdsApi::TestHelpers::AccountApi
+  include SessionHelpers
+
+  context "when FEATURE_FLAG_SAVE_A_PAGE environment variable is not 'enabled'" do
+    context "GET /account/saved-pages/add" do
+      should "return 404" do
+        get :create, params: { page_path: ministry_of_magic_path }
+        assert_response :not_found
+      end
+    end
+  end
+
+  context "when FEATURE_FLAG_SAVE_A_PAGE environment variable is set to 'enabled'" do
+    context "GET /account/saved-pages/add" do
+      context "when logged out" do
+        setup do
+          @stub = stub_account_api_get_sign_in_url(
+            level_of_authentication: "level0",
+            redirect_path: save_page_path(
+              page_path: ministry_of_magic_path,
+            ),
+          )
+        end
+
+        should "redirect the user to the Auth Provider" do
+          with_feature_flag_enabled do
+            get :create, params: { page_path: ministry_of_magic_path }
+            assert_response :redirect
+
+            assert_response :redirect
```

Comment:
You've got this `assert_response :redirect` twice

Reply:
doh, good spot, amended will push shortly.